src/app/navigation/types.ts

```diff
@@ -5,6 +5,7 @@ import { routes } from '../';
 import { AnyAction, AppServices, MiddlewareAPI } from '../types';
 
 export type State = Location & {
+  match?: GenericMatch
```

Comment:
it might help to type this as `AnyMatch`, the two are slightly different

Reply:
This would require chaning more types since we are obtaing `match` from
```ts
export interface LocationChange<M = GenericMatch> {
  location: Location;
  match?: M;
  action: Action;
}
```